pilot/pkg/model/push_context.go

```diff
@@ -252,7 +253,13 @@ type consolidatedDestRules struct {
 	// Map of dest rule host to the list of namespaces to which this destination rule has been exported to
 	exportTo map[host.Name]map[visibility.Instance]bool
 	// Map of dest rule host and the merged destination rules for that host
-	destRule map[host.Name]*config.Config
+	destRule map[host.Name]*consolidateDestRule
+}
+
+// consolidateDestRule represents a dr and from which it is consolidated.
+type consolidateDestRule struct {
```

Comment:
consolidatedDestRules already exists. So it would be very confusing to have this struct. I think DestRuleConsolidationSource may be better. If you agree, change sourceRules -> fromRules?

Reply:
consolidatedDestRules is a wrapper of rules from a single namespace. That one is plural.

How about calling `from`?

Reply:
from makes sense